test/lib/index.test.js

```diff
@@ -2,15 +2,50 @@
 
 const fs = require('fs');
 const expect = require('chai').expect;
+const proxyquire = require('proxyquire');
 
-const libingester = require('../../lib/index');
+const libingester = proxyquire('../../lib/index', {
+    './verify': {
+        verify_metadata: () => {},
+        verify_manifest_entry: () => {},
+    },
+});
+
+class MockAsset extends libingester.BaseAsset {
+    fails_with_error(err) { this._err = err; }
+    set_dependent_assets(dependents) { this._dependents = dependents; }
+    get_dependent_assets() {
+        if (this._dependents) {
+            return this._dependents.map(asset => asset.asset_id);
+        } else {
+            return [];
+        }
+    }
+    _process() {
+        if (this._err) {
+            return Promise.reject(this._err);
+        } else {
+            return Promise.resolve();
+        }
+    }
+}
+
+function rmrf (dir) {
+    fs.readdirSync(dir).forEach(file => fs.unlinkSync(`${dir}/${file}`));
```

Comment:
`fs-extra` could be easier but not a blocker (https://github.com/jprichardson/node-fs-extra/blob/master/lib/remove/__tests__/remove-dir.test.js)

Reply:
IMO the fewer dependencies the better